sophus/rxso2.hpp

```diff
@@ -594,10 +577,7 @@ class RxSO2 : public RxSO2Base<RxSO2<Scalar_, Options>> {
   ///
   SOPHUS_FUNC static Transformation hat(Tangent const& a) {
     Transformation A;
-    // clang-format off
-    A << a(1), -a(0),
-         a(0),  a(1);
-    // clang-format on
+    A << a.reverse(), Vector2<Scalar>{-a.x(), a.y()};
```

Comment:
Not sure whether this is more concise.

The old implementation is definitely more readable. I'd only thing we should change it if the new one proves to be more efficient somehow.